File: pkg/client/acls/acls.go
package acls

import (
	"context"
	"net/http"

	"github.com/ivanzzeth/remote-signer/pkg/client/internal/transport"
)

// IPWhitelistResponse is the read-only IP whitelist config (admin only).
type IPWhitelistResponse struct {
	Enabled         bool     `json:"enabled"`
	AllowedIPs      []string `json:"allowed_ips"`
	TrustProxy      bool     `json:"trust_proxy"`
	TrustedProxies  []string `json:"trusted_proxies"`
}

// Service provides read-only ACL operations (admin only).
type Service struct {
	transport *transport.Transport
}

// NewService creates a new ACLs service.
func NewService(t *transport.Transport) *Service {
	return &Service{transport: t}
}

// GetIPWhitelist returns the IP whitelist configuration (admin only).
func (s *Service) GetIPWhitelist(ctx context.Context) (*IPWhitelistResponse, error) {
	var resp IPWhitelistResponse
	err := s.transport.Request(ctx, http.MethodGet, "/api/v1/acls/ip-whitelist", nil, &resp, http.StatusOK)
	if err != nil {
		return nil, err
	}
	return &resp, nil
}

File: pkg/rs-client/src/acls.rs
use reqwest::Method;
use serde::{Deserialize, Serialize};

use crate::error::Error;
use crate::transport::transport::Transport;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct IpWhitelistResponse {
    pub enabled: bool,
    pub allowed_ips: Vec<String>,
    pub trust_proxy: bool,
    pub trusted_proxies: Vec<String>,
}

#[derive(Clone)]
pub struct Service {
    transport: Transport,
}

impl Service {
    pub fn new(transport: Transport) -> Self {
        Self { transport }
    }

    pub fn get_ip_whitelist(&self) -> Result<IpWhitelistResponse, Error> {
        self.transport
            .request_json(Method::GET, "/api/v1/acls/ip-whitelist", Option::<&()>::None, Some(&[200]))
    }
}
